### DISCUSSION TITLE
Disable Alert/Alarm Only By Entering Data?

### DISCUSSION THREAD

**Discussion by Axehaft** [NightscoutFoundation/xDrip#2238] [Q&A]
**Problem**:  a low (or high) alarm goes off in the night.  The person does not wake, but instead, in his sleep, just silences the alarm.

**Possible Solution**:  Instead of making it easy to silence the alarm, would it be possible to make it so the only way an alarm could be silenced is by entering a blood/carb/units number?  

This would be helpful especially for those T1D's who don't naturally wake when low at night.

Any other ideas about how to deal with this type of situation would be appreciated.

**Comment by Navid200** [comment]
This is extremely personal and a matter of preference.  It's like a choice for wearing a clothing style.  One size doesn't fit all.
There are constant requests to make it easier to snooze alerts.  That's where the majority is.  Although, I cannot claim that I have a valid poll analysis.  My statement is only based on my continuous presence in the xDrip facebook group.

Please note that there are individuals who are concerned about waking their partners at night due to an alert.  So, they wouldn't want the snooze process to have any new obstacles added to it.

I have to be careful not to make any advice that could be mistaken as a medical advice.
So, I suggest that you speak to your health care professional.  Mention to them that you silence your alert and go back to sleep instead of treating yourself.
Ask them for advice.

I hope the following is not offensive.  It's not meant to be.
The driver of a car drives such that his car damages neighbor's car.  He contacts the car manufacturer and asks them to modify the car such that it wouldn't damage the neighbor's car.
The manufacturer can entertain that request.  But, I don't think that would be the best solution for the problem.

**Reply by Axehaft** [reply]
Not offended and I understand where you are coming from ...

I was not looking for medical advice.  I'm looking at the technology (which we've been using daily for 4 yrs) and how baking in a certain option might be helpful in specific circumstances.

If you want to continue with the car analogy, it would be more like this:  A car has a drivers seat that is put in a specific place and height to accommodate *average* drivers.  The seat is not adjustable and causes some to have issues controlling the car.  So instead, the car manufacturer makes the seat adjustable, so that drivers of all sorts can adjust it to work best for them.  That is what I'm looking at.

Thanks.

**Reply by Navid200** [reply]
You have got the attention of the head developer.  So, you can continue with him.

**Comment by jamorham** [comment]
I don't see any reason why we couldn't add an optional feature to require a challenge to cancel an alarm for those users who have difficulty waking up. The question really is how hard does the challenge need to be and under what circumstances should it be applied and what else should/could we do?

For example, would a simple press these buttons in this order be sufficient or are we saying that a user might wake up enough to perform that function without responding to the alert? Do we need to make them do math sums?

Should the alert in this situation be more intense such as giant flashing text, YOU ARE LOW with additional spoken messages and turning the volume up to maximum?

I'm thinking this should be very optional and with time of day trigger. It might also be possible to do something smart such as detecting charging after a certain time of day to decide that the user is going to bed etc.

Really I'm just asking for users to describe how app changes could assist with this scenario to determine what if any features could be put in place that would meet their needs.

**Reply by Navid200** [reply]
There are already too many customizations in xDrip.  As a result, xDrip has become a challenge for most users to use.
In my opinion, we need to strike a balance when deciding what new setting should be added.
Otherwise, there will be no end.

Even if we add this feature, there is no guarantee the single individual who has asked for it will solve his/her problem.  But, the setting will remain and there will be resistance in removing it when I ask "why do we need this"?

We have no clean-up strategy.  We keep adding settings.  But, we never retire any.

**Comment by Navid200** [comment]
While you are considering different options, may I confirm that you know you can choose how the alert sounds?  
You can choose a custom sound.  You can use an mp3 file that would be used as the sound for your alert.

**Reply by Axehaft** [reply]
Yes.  We are very familiar with most xDrip functions and use many of them fairly extensively including our own custom Klaxon sound that we use to indicate a lows.  Other alerts use other sounds.

**Reply by Navid200** [reply]
Very well.

Not that I have any authority.  But, if I did, I would ask:
Let's say the feature you are asking is added.  So, snoozing the alert becomes harder with a setting.  What's going to stop you from disabling that setting when the alert is triggered in the middle of the night?

**Reply by Axehaft** [reply]
For specific alarms, some action is immediately required.  ie.  a reading of 4.0 or lower means carbs are needed.  If the alarm would go off every minute until a carb amount is entered into xDrip (something that requires multiple steps to accomplish), someone who has gotten into the habit of simply silencing the alarm will have to actually look at his phone and realize that he is low and action must be taken.  Making the process of silencing the alarm harder which should increase the chances of the person waking up enough to take care of the issue.

It would just be a more complicated step to take to make the irritating sound in the middle of the night go away.

**Reply by Navid200** [reply]
When the alarm goes off, if you went to the kitchen and ate something, you wouldn't need any change to xDrip, right?
The problem is that you snooze the alert and go back to sleep.

If we add the feature you are asking for, we cannot force it on everyone.   We need to add a setting that allows the user to enable or disable the feature.

My question is now when the alert goes off, you will have two options.  
1- Eat something so that your blood glucose rises.  
2- Just disable the setting so that you can snooze the alert, as you did before, and go back to sleep.

How do you know if you will take option 1 and not option 2?
Obviously you are not in 100% control.  If you were, you would just eat something; you wouldn't ask xDrip to change.
Considering you are not in 100% control, how do you know which option you will take?

**Reply by Axehaft** [reply]
I can't speak for others, but this would help me.

The idea I am suggesting is to make the process of cancelling an alarm harder rather than easier which might increase the chance of a heavy sleeper actually having to wake to get the noise to stop.  Once truly awake, presumably the issue causing the alarm could be taken care of.

Of course this should simply be an option within xDrop and not forced on everyone (like most of the options.)  It could be added to the alarm creation page. 

Thanks.

**Reply by jamorham** [reply]
@Axehaft the idea about requiring carbs input is interesting but my concern is that this is something which can be learned and at a low level of consciousness can be entered without waking up (eg just tell it 10g and go back to sleep).  I'm thinking it needs to be a challenge like, what is 3 + 4 where the numbers change but the sum is easy if at a higher level of consciousness. We could allow snoozing the alarm for only short periods until this is complete (eg 5 minute snooze) so that it doesn't traumatize the user with the alarm blaring while they try and sleepily solve maths sums while low...

Also I think this of course should not be a default feature but enabled by users who find themselves not waking up. I think use of speech could be good here also to tell them what is required or highlight the alarm.

**Reply by Axehaft** [reply]
The math challenge would work much the same way I think.  Anything that forces the sleeper to actually wake up to complete.

**Reply by Navid200** [reply]
We are talking about someone who is not fully themselves because they are half sleep.  And we are hoping that if we make snoozing the alert harder, they will eat because they have no other choice.
But, it seems to me we are ignoring a plethora of other choices available to this unpredictable individual when the alert goes off.  

What will force the sleeper not to disable the setting so that he can snooze the alert easily as before and go back to sleep?
Are we going to also have xDrip ask a math question before allowing that particular setting to be changed?

What will stop the sleeper from just shutting off the phone and going back to sleep?

What if the individual just walks to another quiet room away from the ringing phone and sleeps there?

What if the sleeper goes to the kitchen to eat but by the time he gets there, he just falls sleep in the kitchen instead of eating?

All of those alternatives allow the individual, who does not appreciate the severity of diabetes at the time because they are half sleep, to go back to sleep without eating. 


I appreciate the problem.
With all due respect, I don't think the suggestion is a real practical solution.
